From: Maximilian Weigand <mweigand@mweigand.net>
Date: Fri, 23 Feb 2024 19:52:54 +0100
Subject: Pinenote: modify frequencies of ACLK_TOP_HIGH and CLK_TOP_LOW to
 match android

Previously, we defaulted to 200 MHz (TOP_LOW) and 300 MHz (TOP_HIGH).
However, the rockchip android kernel uses 500 and 400 MHz, respectively.
---
 arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
index 45567a6..a501d43 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
@@ -285,10 +285,20 @@ vdda_0v9: vdda-0v9 {
 
 &cru {
 		// assigned-clocks = <&pmucru CLK_RTC_32K>, <&cru PLL_GPLL>, <&pmucru PLL_PPLL>, <&cru ACLK_RGA_PRE>, <&cru HCLK_RGA_PRE>;
-		assigned-clocks = <&pmucru CLK_RTC_32K>, <&cru PLL_GPLL>, <&pmucru PLL_PPLL>, <&cru ACLK_RGA_PRE>, <&cru CPLL_250M>, <&cru HCLK_RGA_PRE>, <&cru HCLK_EBC>, <&cru HCLK_JENC>;
+		assigned-clocks = <&pmucru CLK_RTC_32K>,
+  			<&cru ACLK_TOP_HIGH>, <&cru ACLK_TOP_LOW>,
+			<&cru PLL_GPLL>, <&pmucru PLL_PPLL>,
+ 			<&cru ACLK_RGA_PRE>, <&cru CPLL_250M>,
+			<&cru HCLK_RGA_PRE>, <&cru HCLK_EBC>,
+		    <&cru HCLK_JENC>;
 		// assigned-clock-rates = <32768>, <1200000000>, <200000000>, <300000000>, <300000000>;
 		// assigned-clock-rates = <32768>, <1200000000>, <200000000>, <250000000>, <250000000>, <250000000>, <250000000>, <250000000>;
-		assigned-clock-rates = <32768>, <1200000000>, <200000000>, <300000000>, <300000000>, <300000000>, <300000000>, <300000000>;
+		assigned-clock-rates = <32768>,
+				<500000000>, <400000000>,
+				 <1200000000>, <200000000>,
+				 <300000000>, <300000000>,
+				 <300000000>, <300000000>,
+				 <300000000>;
 		// stronger artifacting, I suspend instabilities due to too high clock frequencies
 		// assigned-clock-rates = <32768>, <1200000000>, <200000000>, <333400000>, <333400000>, <333400000>, <333400000>, <333400000>;
 		assigned-clock-parents = <&pmucru CLK_RTC32K_FRAC>;
